contracts/src/Staking.sol

```diff
@@ -0,0 +1,535 @@
+
+        uint256 oldAmount = fixedStakeAmount;
+        fixedStakeAmount = proposal.value;
+        delete pendingFixedStakeAmountChange;
+        emit FixedStakeAmountChanged(oldAmount, proposal.value);
+    }
+
+    /*
+     * @notice Execute a pending withdraw delay change
+     */
+    function executeWithdrawDelayChange() external {
+        ConfigProposal memory proposal = pendingWithdrawDelayChange;
+        if (proposal.executableAt == 0) revert NoProposalExists();
+        if (block.timestamp < proposal.executableAt) revert ProposalNotExecutable();
+
+        uint256 oldDelay = withdrawDelay;
+        withdrawDelay = proposal.value;
+        delete pendingWithdrawDelayChange;
+        emit WithdrawDelayChanged(oldDelay, proposal.value);
+    }
+
+    /*
+     * @notice Execute pending validator changes
+     */
+    function executeValidatorChanges() external {
+        ValidatorProposal memory proposal = pendingValidatorChanges;
+        if (proposal.executableAt == 0) revert NoProposalExists();
+        if (block.timestamp < proposal.executableAt) revert ProposalNotExecutable();
+
+        for (uint256 i = 0; i < proposal.validators.length; i++) {
+            isValidator[proposal.validators[i]] = proposal.isRegistration[i];
+            emit ValidatorUpdated(proposal.validators[i], proposal.isRegistration[i]);
+        }
+
+        delete pendingValidatorChanges;
+    }
+
+    // ============================================================
+    // EXTERNAL FUNCTIONS - TOKEN RECOVERY (OWNER ONLY)
+    // ============================================================
+
+    /*
+     * @notice Recover accidentally sent tokens
+     * @param token The token address to recover
+     * @param to The address to send recovered tokens to
+     */
+    function recoverTokens(address token, address to) external onlyOwner {
+        if (to == address(0)) revert InvalidAddress();
+
+        uint256 recoverable;
+        if (token == address(safeToken)) {
+            uint256 balance = safeToken.balanceOf(address(this));
+            recoverable = balance - totalStakedAmount - totalPendingWithdrawals;
+        } else {
+            recoverable = IERC20(token).balanceOf(address(this));
+        }
+
+        if (recoverable == 0) revert InsufficientRecoverableAmount();
+
+        emit TokensRecovered(token, to, recoverable);
+
+        IERC20(token).safeTransfer(to, recoverable);
+    }
+
+    // ============================================================
+    // VIEW FUNCTIONS - STAKING QUERIES
+    // ============================================================
+
+    /*
+     * @notice Get the voting power of a validator
+     * @param validator The validator address
+     * @return The number of votes
+     */
+    function getVotingPower(address validator) external view returns (uint256) {
+        if (!isValidator[validator]) return 0;
+        return totalStakes[validator] / fixedStakeAmount;
```

Comment:
Maybe it makes sense to move the `fixedStakeAmount` to the consensus contract ... but not sure. Currently it feels under utilized here (only in these view methods).